### PR TITLE
Fix regression on oauth2 access_token forwarding from gateway to microservices

### LINKED ISSUE #7419: Regression on oauth2 access_token forwarding from gateway to microservices
##### **Overview of the issue**

In a micro-service deployment, the access_token is no longer forwarded from the gateway to the microservices.

##### **Reproduce the error**

Generate a gateway and a microservice  with some entity
Add a @Secured annotation on an endpoint the entity Ressource Controller => it doesn't work as there is no Spring Security Context

##### **Suggest a Fix**

The problem is caused by the recreation of the OAuth2Authentication by com.mycompany.oauth2gateway.service.UserService.getUserFromAuthentication(OAuth2Authentication)
used in a gateway since #7106 apparently

This code doesn't preserve the OAuth2AuthenticationDetails that store the access_token value
and com.mycompany.oauth2gateway.gateway.TokenRelayFilter fails to extract (and forward it)

I will produced a PR to fix this shortly

##### **JHipster Version(s)**

```
oauth-2-multigateway@0.0.0 C:\travail\workspaces_oxygen\jhipster\masterOauth2\gateway
`-- generator-jhipster@5.0.0-beta.0

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.oauth2gateway",
      "nativeLanguage": "en",
      "microservicePath": "..\\ms"
    },
    "jhipsterVersion": "5.0.0-beta.0",
    "baseName": "oauth2multigateway",
    "packageName": "com.mycompany.oauth2gateway",
    "packageFolder": "com/mycompany/oauth2gateway",
    "serverPort": "8080",
    "authenticationType": "oauth2",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": "eureka",
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "angularX",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "gateway",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en"
    ],
    "reactive": false
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

git version 2.16.2.windows.1

node: v8.9.4

npm: 5.4.0

yeoman: 1.8.5

yarn: 1.5.1

Docker version 17.10.0-ce, build f4ffd25

docker-compose version 1.16.1, build 6d1ac219

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

n/a

##### **Browsers and Operating System**

Windows 10

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by farrault**
- Please make sure the below checklist is followed for Pull Requests.

- [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [ ] Tests are added where necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/server/templates/src/main/java/package/service/UserService.java.ejs [hunk 1]
```diff
@@ -489,6 +489,7 @@ public class UserService {
      */
     @SuppressWarnings("unchecked")
     public UserDTO getUserFromAuthentication(OAuth2Authentication authentication) {
+        Object oauth2AuthenticationDetails = authentication.getDetails(); // should be a OAuth2AuthenticationDetails
```
Comment (mraible) [inline generators/server/templates/src/main/java/package/service/UserService.java.ejs]: Can you please change this to "an OAuth2AuthenticationDetails"

File: generators/server/templates/src/main/java/package/service/UserService.java.ejs [hunk 2]
```diff
@@ -504,6 +505,7 @@ public class UserService {
 
         UsernamePasswordAuthenticationToken token = getToken(details, user, grantedAuthorities);
         authentication = new OAuth2Authentication(authentication.getOAuth2Request(), token);
+        authentication.setDetails(oauth2AuthenticationDetails); // must be present in a Gateway for TokenRelayFilter
```
Comment (mraible) [inline generators/server/templates/src/main/java/package/service/UserService.java.ejs]: Please change to "must be present in a gateway for TokenRelayFilter to work"

**Comment by deepu105** [comment]
Please change to a better title, this is a community project and hence we need to make sure people can understand a PR from title

**Comment by jdubois** [comment]
Great work @farrault ! Congrats for your 1st contribution to the project :-)

**Pull request merged**